libs/ledger-live-common/src/modularDrawer/hooks/useCurrenciesUnderFeatureFlag.ts

```diff
@@ -0,0 +1,154 @@
+  const lineaSepolia = useFeature("currencyLineaSepolia");
+  const blast = useFeature("currencyBlast");
+  const blastSepolia = useFeature("currencyBlastSepolia");
+  const scroll = useFeature("currencyScroll");
+  const scrollSepolia = useFeature("currencyScrollSepolia");
+  const icon = useFeature("currencyIcon");
+  const ton = useFeature("currencyTon");
+  const etherlink = useFeature("currencyEtherlink");
+  const zksync = useFeature("currencyZkSync");
+  const zksyncSepolia = useFeature("currencyZkSyncSepolia");
+  const mantra = useFeature("currencyMantra");
+  const xion = useFeature("currencyXion");
+  const zenrock = useFeature("currencyZenrock");
+  const sonic = useFeature("currencySonic");
+  const sonicBlaze = useFeature("currencySonicBlaze");
+  const sui = useFeature("currencySui");
+  const mina = useFeature("currencyMina");
+  const babylon = useFeature("currencyBabylon");
+  const seiNetworkEvm = useFeature("currencySeiNetworkEvm");
+  const berachain = useFeature("currencyBerachain");
+  const hyperevm = useFeature("currencyHyperevm");
+  const canton = useFeature("currencyCantonNetwork");
+  const kaspa = useFeature("currencyKaspa");
+  const core = useFeature("currencyCore");
+  const ethereumHoodi = useFeature("currencyEthereumHoodi");
+
+  const featureFlaggedCurrencies: Partial<Record<CryptoCurrencyId, Feature<unknown> | null>> = {
+    aptos,
+    aptos_testnet: aptosTestnet,
+    axelar,
+    stargaze,
+    secret_network: secretNetwork,
+    umee,
+    desmos,
+    dydx,
+    onomy,
+    sei_network: seiNetwork,
+    quicksilver,
+    persistence,
+    avalanche_c_chain: avaxCChain,
+    stacks,
+    optimism,
+    optimism_sepolia: optimismSepolia,
+    arbitrum,
+    arbitrum_sepolia: arbitrumSepolia,
+    rsk,
+    bittorrent,
+    energy_web: energyWeb,
+    astar,
+    metis,
+    boba,
+    moonriver,
+    velas_evm: velasEvm,
+    syscoin,
+    internet_computer: internetComputer,
+    telos_evm: telosEvm,
+    sei_network_evm: seiNetworkEvm,
+    berachain,
+    hyperevm,
+    coreum,
+    polygon_zk_evm: polygonZkEvm,
+    polygon_zk_evm_testnet: polygonZkEvmTestnet,
+    base,
+    base_sepolia: baseSepolia,
+    klaytn,
+    injective,
+    vechain,
+    casper,
+    neon_evm: neonEvm,
+    lukso,
+    linea,
+    ton,
+    linea_sepolia: lineaSepolia,
+    blast,
+    blast_sepolia: blastSepolia,
+    scroll,
+    scroll_sepolia: scrollSepolia,
+    icon,
+    etherlink,
+    zksync,
+    zksync_sepolia: zksyncSepolia,
+    mantra,
+    xion,
+    zenrock,
+    sonic,
+    sonic_blaze: sonicBlaze,
+    sui,
+    mina,
+    babylon,
+    canton_network: canton,
+    kaspa,
+    core,
+    ethereum_hoodi: ethereumHoodi,
+  };
+
+  const deactivatedCurrencyIds = new Set(
+    mock
+      ? [] // mock mode: all currencies are available for playwright tests
+      : Object.entries(featureFlaggedCurrencies)
+          .filter(([, feature]) => !feature?.enabled)
+          .map(([id]) => id),
+  );
```

Comment:
You problaby want to memo this, especially given it's used in dependencies array in mobile and desktop